docs/docs/reference/metaprogramming/toc.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: doc-page
+title: "Overview"
+---
+
+The following pages introduce the redesign of metaprogramming in Scala. They
+introduce the following fundamental facilities:
+
+1. [Inline](./inline.html) `inline` is a new soft-modifier that guarantees that
+   a definition will be inlined at the point of use. The primary motivation
+   behind inline is to reduce the overhead behind function calls and access to
+   values. The expansion will be performed by the Scala compiler during the
+   `Typer` compiler phase. However, as opposed to inline in other ecosystems,
+   inlining is not merely a request to the compiler but for Scala it is a
+   _command_. The reason is that inlining in Scala can driver other compile-time
+   operations too, like inline pattern matching (enabling type-level
+   programming), macros (enabling compile-time, generative, metaprogramming) and
+   runtime code generation (multi-stage programming). In this section we
+   describe up to inline pattern matching describing the basic constructs.
+
+2. [Macros](./macros.html) Macros are built on two well-known fundamental
+   operations: quotation and splicing.  Quotation is expressed as `'{...}` for
+   expressions (both forms are equivalent) and as `'[...]` for types. Splicing
+   is expressed as `${ ... }`. Whereas inlining is driven completely by the
+   language level features of scala (pattern matching, inline vals and
+   definitions), macros enable is synthesize/compute code at will treating code
+   values as first class citizens and splicing them together independently.
+   Here, we move towards _domain-specific_ metaprogramming.
+
+3. [Staging](./staging.html) Macros can be seen as distinct phase while
```

Comment:
This description does not correspond with the contents of `staging.md`

Reply:
True. Will improve staging.